certbot/tests/lock_test.py

```diff
@@ -91,27 +99,36 @@ def test_removed(self):
         lock_file.release()
         self.assertFalse(os.path.exists(self.lock_path))
 
-    @mock.patch('certbot.compat.fcntl.lockf')
-    def test_unexpected_lockf_err(self, mock_lockf):
+    def test_unexpected_lockf_or_locking_err(self):
+        if POSIX_MODE:
+            mocked_function = 'certbot.lock.fcntl.lockf'
+        else:
+            mocked_function = 'certbot.lock.msvcrt.locking'
         msg = 'hi there'
-        mock_lockf.side_effect = IOError(msg)
-        try:
-            self._call(self.lock_path)
-        except IOError as err:
-            self.assertTrue(msg in str(err))
-        else:  # pragma: no cover
-            self.fail('IOError not raised')
-
-    @mock.patch('certbot.lock.os.stat')
-    def test_unexpected_stat_err(self, mock_stat):
+        with mock.patch(mocked_function) as mock_lock:
+            mock_lock.side_effect = IOError(msg)
+            try:
+                self._call(self.lock_path)
+            except IOError as err:
+                self.assertTrue(msg in str(err))
+            else:  # pragma: no cover
+                self.fail('IOError not raised')
+
+    def test_unexpected_os_err(self):
+        if POSIX_MODE:
+            mock_function = 'certbot.lock.os.stat'
+        else:
+            mock_function = 'certbot.lock.msvcrt.locking'
+        # Neither errno.ENOENT nor errno.EACCES that are expected and have a specific logic.
```

Comment:
I'm not sure I understand this comment. Are we trying to say something like:

> The only expected errno are ENOENT and EACCES

If so, can we rephrase the comment to something like that?

Reply:
Rephrased.